packages/rps/src/redux/game/reducer.ts

```diff
@@ -57,203 +37,92 @@ const localReducer: Reducer<LocalState> = (
   state: LocalState = emptyLocalState,
   action: GameAction
 ) => {
-  switch (action.type) {
-    case 'GotAddressFromWallet':
-      return handleGotAddressFromWallet(state, action);
-    case 'UpdateProfile':
-      return handleUpdateProfile(state, action);
-    case 'NewOpenGame':
-      return handleNewOpenGame(state, action);
-    case 'JoinOpenGame':
-      return handleJoinOpenGame(state, action);
-    case 'CreateGame':
-      return handleCreateGame(state, action);
-    case 'CancelGame':
-      return handleCancelGame(state, action);
-    case 'GameJoined':
-      return handleGameJoined(state, action);
+  let newState = state;
+  if (
+    action.type === 'Resign' &&
+    state.type !== 'Empty' &&
+    state.type !== 'NeedAddress' &&
+    state.type !== 'Lobby' &&
+    state.type !== 'WaitingRoom' &&
+    state.type !== 'CreatingOpenGame'
+  ) {
+    newState = resigned(state, action.iResigned);
```

Comment:
Might be nicer to just `return resigned(state, action.iResigned)` here - as someone reading the code, that's nice, because I know that's it and don't have to scroll down to the bottom to see if anything else happens to `newState`. Also removes a variable and a couple of lines 😛.